sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -585,8 +539,31 @@ public ServiceBusReceivedMessage receiveDeferredMessage(long sequenceNumber, Str
      * @throws IllegalArgumentException if {@code lockToken} is an empty value.
      * @throws IllegalStateException if the receiver is a session receiver.
      */
-    public OffsetDateTime renewMessageLock(String lockToken) {
-        return asyncClient.renewMessageLock(lockToken).block(operationTimeout);
+    public OffsetDateTime renewMessageLock(ServiceBusReceivedMessage message) {
+        return asyncClient.renewMessageLock(message).block(operationTimeout);
+    }
+
+    /**
+     * Starts the auto lock renewal for a message with the given lock.
+     *
+     * @param message The {@link ServiceBusReceivedMessage} to perform auto-lock renewal.
+     * @param maxLockRenewalDuration Maximum duration to keep renewing the lock token.
+     * @param onError A function to call when an error occurs during lock renewal.
+     *
+     * @throws NullPointerException if {@code lockToken} or {@code maxLockRenewalDuration} is null.
+     * @throws IllegalArgumentException if {@code lockToken} is an empty string.
+     * @throws IllegalStateException if the receiver is a session receiver or the receiver is disposed.
+     */
+    public void renewMessageLock(ServiceBusReceivedMessage message, Duration maxLockRenewalDuration,
+        Consumer<Throwable> onError) {
+        final String lockToken = message != null ? message.getLockToken() : "null";
```

Comment:
What is the reason of assigning "null" string to `lockToken` . Shouldn't  we throw NPE if it is `null`.

Reply:
It'll get thrown later when we call into asyncClient. This is so the function invocation resolves correctly.